src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -1250,6 +1250,11 @@ private bool haveSameMethodDefinition(CORINFO_METHOD_STRUCT_* methHnd1, CORINFO_
             return meth1.GetTypicalMethodDefinition() == meth2.GetTypicalMethodDefinition();
         }
 
+        private CORINFO_CLASS_STRUCT_* getTypeDefinition(CORINFO_CLASS_STRUCT_* type)
+        {
+            return ObjectToHandle(HandleToObject(type).GetTypeDefinition());
```

Comment:
does it need an assert that the input must be generic? (coreclr's impl does validate that)

Reply:
Added a check for `HasInstantiation` in 9dacde374ad1398d7c9c87f783d6f854ddc54843.